src/net_processing.cpp

```diff
@@ -3329,7 +3329,13 @@ bool ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRec
         pfrom.fSentAddr = true;
 
         pfrom.vAddrToSend.clear();
-        std::vector<CAddress> vAddr = connman->GetAddresses();
+        std::vector<CAddress> vAddr;
+        if (pfrom.HasPermission(PF_ADDR)) {
+            vAddr = connman->GetAddresses(nullopt);
+        } else {
+            Network peer_network = pfrom.addr.GetNetwork();
+            vAddr = connman->GetAddresses(peer_network);
```

Comment:
Merge these two lines? No need for a local `peer_network` variable (the compiled code will be exactly the same, but creating a local variable that won't be used again signals the wrong intent in the code)